pkg/common/cns-lib/volume/manager.go

```diff
@@ -1776,6 +1785,13 @@ func (m *defaultManager) createSnapshotWithImprovedIdempotencyCheck(ctx context.
 				// Don't return if CreateSnapshot details can't be stored.
 				log.Warnf("failed to store CreateSnapshot details with error: %v", err)
 			}
+		} else {
+			// store task details into snapshotTaskMap
+			var taskDetails createSnapshotTaskDetails
+			taskDetails.task = createSnapshotsTask
+			taskDetails.expirationTime = time.Now().Add(time.Hour * time.Duration(
+				defaultOpsExpirationTimeInHours))
+			snapshotTaskMap[snapshotName] = &taskDetails
```

Comment:
Is it better to use "instanceName" rather than just "snapshotName"?

Reply:
It should be OK to use `snapshotName` here as it is also unique. But, it is better to use one of them as the key of `createSnapshotsTask` and `CnsVolumeOperationDetails` instance for CreateSnapshot operation. Good catch!